### PR TITLE
Create addresses model

### PR DESCRIPTION
**Pull request by hamza-yusuff**
## Notion ticket link

[https://www.notion.so/uwblueprintexecs/Create-addresses-model-11e335fdac594c7bbbebdde90723294e]

## Implementation description
* Created address,py file inside models
* Created the model for addresses table inside the address.py file
* Ran migration and upgraded the database

## Steps to test
1. Access the database - docker exec -it cas_py_backend /bin/bash
2. Run the command - SELECT * FROM addresses;
3. If you see the table addresses with the necessary columns, then the change should have been made.

### CODE REVIEW

File: backend/python/app/models/address.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    id = db.Column(db.Integer, primary_key=True, nullable=False)
+    street_address = db.Column(db.String, nullable=False)
+    city = db.Column(db.String, nullable=False)
+    postal_code = db.Column(db.String, nullable=False)
+    latitude = db.Column(db.Numeric(8, 6), nullable=False)
+    longitude = db.Column(db.Numeric(9, 6), nullable=False)
+
+    def to_dict(self, include_relationships=False):
+        # define the entities table
```
Comment (jennchenn) [inline backend/python/app/models/address.py:18]: nit: remove this comment or change to `addresses` table

File: backend/python/app/models/address.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        cls = type(self)
+
+        mapper = inspect(cls)
+        formatted = {}
+        for column in mapper.attrs:
+            field = column.key
+            attr = getattr(self, field)
+            # if it's a regular column, extract the value
```
Comment (jennchenn) [inline backend/python/app/models/address.py:26]: nit: remove the comments here and on the lines below (they're already in the other model files, they don't need to be repeated in all new model files)

File: backend/python/app/models/address.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+
+class Address(db.Model):
+    __tablename__ = "addresses"
+
+    id = db.Column(db.Integer, primary_key=True, nullable=False)
+    street_address = db.Column(db.String, nullable=False)
+    city = db.Column(db.String, nullable=False)
+    postal_code = db.Column(db.String, nullable=False)
+    latitude = db.Column(db.Numeric(8, 6), nullable=False)
```
Comment (jennchenn) [inline backend/python/app/models/address.py:14]: can you make these `nullable=True` instead? you will need to adjust the migration file as well

**Review by jennchenn — CHANGES_REQUESTED** [review]
tested it out and looks good!

File: backend/python/app/models/address.py (hunk 3 above)
Comment (jennchenn) [reply backend/python/app/models/address.py]: ahh sorry github did not tie my comment to the line properly i meant to comment only on the latitude and longitude fields 🙏

File: backend/python/app/models/address.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+        mapper = inspect(cls)
+        formatted = {}
+        for column in mapper.attrs:
+            field = column.key
+            attr = getattr(self, field)
+
+            if isinstance(column, ColumnProperty):
+                formatted[field] = attr
+
```
Comment (jennchenn) [inline backend/python/app/models/address.py:29]: nit: remove empty line

File: backend/python/app/models/address.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+            elif include_relationships:
+
```
Comment (jennchenn) [inline backend/python/app/models/address.py:31]: nit: remove empty line

**Review by jennchenn — CHANGES_REQUESTED** [review]

**Comment by hamza-yusuff** [comment]
Sorrry about the PR name, completely forgot how I am supposed to do the naming ?

**Comment by jennchenn** [comment]
> Sorrry about the PR name, completely forgot how I am supposed to do the naming ?

you can either use the ticket name/just something that is more descriptive of what you're doing - the PR name is what is going to show up in the commit history in the main branch so it would be good to just have it as descriptive as possible :)

**Review by jennchenn — APPROVED** [review]
lgtm!!

**Pull request merged**